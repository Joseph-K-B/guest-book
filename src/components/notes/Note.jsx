import React from 'react';
import useAuth from '../../hooks/useAuth';

function Note({ note: {message} }) {
    const {user} = useAuth();
    return(
        <><h2>{user.username}</h2><p>{message}</p></>
    )
}

export default Note;