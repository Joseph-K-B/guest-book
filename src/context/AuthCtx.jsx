import React, { createContext, useState } from "react"

const authCtx = createContext();

const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  // This would probably be the sign up function 
  // b/c it includes both username and email
  // 
  const login = (username, email, password) => {
    const authConfirmed = (
      username === process.env.REACT_APP_AUTH_USER &&
      email === process.env.REACT_APP_AUTH_EMAIL &&
      password === process.env.REACT_APP_AUTH_PASSWORD
      )
      if(authConfirmed) 
        setUser({ username, email });
      return authConfirmed;
  };

  const logout = (cb) => {
    setUser(null);
    // cb();
  }

  return (
    <authCtx.Provider value={{user, setUser, login, logout}}>
      {children}
    </authCtx.Provider>
  );
};

export {authCtx, AuthProvider};