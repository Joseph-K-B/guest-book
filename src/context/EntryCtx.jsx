import { createContext, useContext, useState } from "react";

const NoteCtx = createContext();

const NoteProvider = ({children}) => {
    const [note, setNote] = useState('');

    return <NoteCtx.Provider value={{ note, setNote }}>{children}</NoteCtx.Provider>
}

const useNote = () => {
    const ctx = useContext(NoteCtx);

    if(ctx === undefined) {
        throw new Error('useNote hook can only be called within child components of NoteCtx Provider');
    }
    return ctx;
}

export { NoteProvider, useNote}