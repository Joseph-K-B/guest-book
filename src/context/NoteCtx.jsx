import React from 'react';
import { createContext, useContext, useState } from "react";

const NoteCtx = createContext();

const NoteProvider = ({ children }) => {
    const [notes, setNotes] = useState([]);

    return (
        <NoteCtx.Provider value={{ notes, setNotes }}>{children}</NoteCtx.Provider>
    )
}

const useNote = () => {
    const ctx = useContext(NoteCtx);

    if(ctx === undefined) {
        throw new Error('useNote hook can only be called within child components of NoteCtx Provider');
    }
    return ctx;
}

export { NoteProvider, useNote}