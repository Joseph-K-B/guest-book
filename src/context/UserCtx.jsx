import React from 'react';
import { createContext, useContext, useState } from "react"

const UserCtx = createContext();

const UserProvider = ({children}) => {
    const [name, setName] = useState('');

    return <UserCtx.Provider value={{ name, setName }}>{children}</UserCtx.Provider>
}

const useUser = () => {
    const ctx = useContext(UserCtx);

    if(ctx === undefined) {
        throw new Error('useUser custom hook must be called within child components of UserCtx Provider');
    }
    return ctx;
}

export { UserProvider, useUser }