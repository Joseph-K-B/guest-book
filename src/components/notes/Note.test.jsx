import React, { createContext } from 'react';
import { screen, render } from '@testing-library/react';
import { NoteProvider } from '../../context/NoteCtx';
import Note from './Note';
import { authCtx, AuthProvider } from '../../context/AuthCtx';

jest.mock('../../context/AuthCtx'), () => ({
  __esModue: true,
  default: createContext()
 })

const user = {
  username: 'Izzie the Dog',
  email: 'izzie@dog.com',
}


it('renders a note and user ctx', async () => {
    const {container} = render(
    <AuthProvider>
      <authCtx.Provider value={{user}}>      
        <NoteProvider>
          <Note note = {{ message: 'words here'}} />
        </NoteProvider>
      </authCtx.Provider>
    </AuthProvider>
  );

    const guestName = screen.getByText(/Izzie the Dog/i);
    expect(guestName).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});