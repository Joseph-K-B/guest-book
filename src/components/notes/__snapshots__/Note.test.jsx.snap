// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a note left by user 1`] = `
<div>
  <h2>
    Izzie the Dog
  </h2>
  <p>
    words here
  </p>
</div>
`;
